willow-eventhandler/src/main/java/com/nitorcreations/willow/eventhandler/EventPoller.java

```diff
@@ -0,0 +1,231 @@
+  private Logger logger = Logger.getLogger(getClass().getName());
+
+  private Map<String, List<EventHandler>> eventHandlers;
+
+  @Inject
+  private Gson gson;
+
+  @Inject
+  private ExecutorService executorService;
+
+  /**
+   * Initialize the event poller.
+   * 
+   * @param eventHandlers The handlers for different event types.
+   * @param uri The web socket connection URI.
+   */
+  public void initialize(Map<String, List<EventHandler>> eventHandlers, URI uri) {
+    this.eventHandlers = eventHandlers;
+    EventListener listener = new EventListener(uri);
+    executorService.submit(listener);
+    logger.fine("Started event listener");
+  }
+
+  /**
+   * Websocket message listener. Dispatches events to the event handlers.
+   * Autowired to Jetty websocket implementation via annotations.
+   * 
+   * @author mtommila
+   */
+  @WebSocket
+  public class EventListener implements Runnable {
+
+    private Logger logger = Logger.getLogger(getClass().getName());
+
+    private URI uri;
+    private WebSocketClient client;
+    private CountDownLatch connectLatch;
+    private Set<String> messageIds = new HashSet<>();
+
+    private EventListener(URI uri) {
+      this.uri = uri;
+    }
+
+    private synchronized WebSocketClient getClient() {
+      if (this.client == null) {
+        this.client = new WebSocketClient();
+      }
+      return this.client;
+    }
+
+    private synchronized WebSocketClient createClient() {
+      this.client = new WebSocketClient();
+      return this.client;
+    }
+
+    private void connect() throws Exception {
+      logger.info("Connecting event listener");
+      WebSocketClient client = createClient();
+      client.start();
+      client.setAsyncWriteTimeout(5000);
+      client.setConnectTimeout(2000);
+      client.setStopTimeout(5000);
+      ClientUpgradeRequest request = new ClientUpgradeRequest();
+      request.setHeader("Authorization", SSHUtil.getPublicKeyAuthorization(System.getProperty("user.name", "willow")));
+      Future<Session> future = client.connect(this, uri, request);
+      logger.info("Connecting to : " + uri);
+      try {
+        future.get();
+        logger.info("Connected to : " + uri);
+      } catch (Exception e) {
+        logger.log(Level.SEVERE, "Failed to connect event poller with uri " + uri, e);
+        if (client.isRunning()) {
+          client.stop();
+          client.destroy();
+        }
+        throw e;
+      }
+    }
+
+    @OnWebSocketMessage
+    public void messageReceived(Session session, String message) {
+      if (message == null || message.isEmpty()) {
+        return;
+      }
+      EventMessage[] eventMessages = null;
+      List<EventMessage> uniqueEventMessages = new ArrayList<>();
+      try {
+        eventMessages = gson.fromJson(gson.fromJson(message, JsonObject.class).get("data"), EventMessage[].class);
+      } catch (Exception e) {
+        logger.log(Level.INFO, "Failure in unmarshalling event data", e);
```

Comment:
just return instead of handling eventMessages == null after the exception. Makes the control flow easier to understand.